### PR TITLE
bulk updates and deletions

### PR DESCRIPTION
**Pull request by ermyas**
closes issue #6

### CODE REVIEW

**Comment by beloglazov** [comment]
Looks like this PR contains changes introduced in #5, I'd merge #5 first.

File: src/main/scala/com/ibm/couchdb/api/Documents.scala [hunk 1]
```diff
@@ -132,7 +163,11 @@ class Documents(client: Client, db: String, typeMapping: TypeMapping) {
       Res.Error("cannot_delete", "The attachment name is empty").toTask[Res.DocOk]
     else
       client.delete[Res.DocOk](
-        s"/$db/${ obj._id }/$name?rev=${ obj._rev }",
+        s"/$db/${
+          obj._id
+        }/$name?rev=${
+          obj._rev
+        }",
```
Comment (beloglazov) [inline src/main/scala/com/ibm/couchdb/api/Documents.scala]: Is this change and a few above a result of auto-formatting? It seems to me that string literals are easier to read when they are on a single line.

File: src/main/scala/com/ibm/couchdb/api/Documents.scala [hunk 2]
```diff
@@ -47,16 +47,33 @@ class Documents(client: Client, db: String, typeMapping: TypeMapping) {
         CouchDoc[T](obj, types(cl)))
   }
 
+  private def updateBulk[T: upickle.Writer](objs: Seq[CouchDoc[T]]): Task[Seq[Res.DocOk]] = {
```
Comment (beloglazov) [inline src/main/scala/com/ibm/couchdb/api/Documents.scala]: Maybe call it `postBulk`? As it's also used in create, delete, etc.

File: src/main/scala/com/ibm/couchdb/api/Documents.scala [hunk 3]
```diff
@@ -47,16 +47,33 @@ class Documents(client: Client, db: String, typeMapping: TypeMapping) {
...
+    client.post[Req.Docs[T], Seq[Res.DocOk]](
+      s"/$db/_bulk_docs",
+      Status.Created, Req.Docs(objs))
```
Comment (beloglazov) [inline src/main/scala/com/ibm/couchdb/api/Documents.scala:53]: `Status.Created` is only relevant for `createMany`, should be set appropriately for delete and update queries.

File: src/test/scala/com/ibm/couchdb/api/DocumentsSpec.scala [hunk 4]
```diff
@@ -216,6 +228,71 @@ class DocumentsSpec extends CouchDbSpecification {
...
-  }
+    "Bulk update should" >> {
+      val fixes = Seq(fixAlice, fixBob, fixHaile)
+      def change(s: String) = s + "-updated"
+      def create(x: Seq[FixPerson]): Seq[CouchDoc[FixPerson]] = {
+        clear()
+        val newIds = awaitRight(documents.createMany(x)).map(_.id)
+        awaitRight(documents.getMany[FixPerson](newIds)).getDocs
+      }
+      def modify(orig: Seq[CouchDoc[FixPerson]]): Seq[CouchDoc[FixPerson]] = orig.map(x => x applyLens _docPersonName modify change)
```
Comment (beloglazov) [inline src/test/scala/com/ibm/couchdb/api/DocumentsSpec.scala]: Line too long?

File: src/main/scala/com/ibm/couchdb/api/Documents.scala (hunk 3 above)
Comment (ermyas) [inline src/main/scala/com/ibm/couchdb/api/Documents.scala:53]: couch returns the same '201 Created' on all POST requests (creation, updates, deletion) to /_bulk_docs

**Pull request merged**